x-pack/plugins/cloud_security_posture/public/components/fleet_extensions/eks_credentials_form.tsx

```diff
@@ -229,20 +240,77 @@ export const EksCredentialsForm = ({ input, newPolicy, updatePolicy }: Props) =>
   const group = options[awsCredentialsType];
   const fields = getInputVarsFields(input, group.fields);
 
+  // to avoid sending up credential fields unrelated to the new selection
+  // we confirm with the user that we will clear these fields if they switch methods (edit mode only)
+  const [confirmTypeChange, setConfirmTypeChange] = useState<string | false>(false);
+  const onCredentialTypeChange = useCallback(
+    (optionId: string, confirmed?: boolean) => {
+      const fieldsHaveValues = !!fields.find((field) => field.value);
+      if (fieldsHaveValues && !confirmed) {
+        setConfirmTypeChange(optionId); // stored, so the confirm modal can call this function if the user hits OK
+      } else {
+        // clear out credentials when switching methods to avoid persisting unused values.
+        clearInputVarsFields(input, group.fields);
+
+        updatePolicy(
+          getPosturePolicy(newPolicy, input.type, {
+            'aws.credentials.type': { value: optionId },
+          })
+        );
+      }
+    },
+    [fields, group.fields, input, newPolicy, updatePolicy]
+  );
+
+  const onCancelTypeChange = useCallback(() => {
+    setConfirmTypeChange(false);
+  }, []);
+
+  const onConfirmTypeChange = useCallback(() => {
+    if (confirmTypeChange) {
+      onCredentialTypeChange(confirmTypeChange, true);
+      setConfirmTypeChange(false);
+    }
+  }, [confirmTypeChange, onCredentialTypeChange]);
+
   return (
     <>
+      {confirmTypeChange && (
```

Comment:
I'm not a fan of that option to prompt the user it will clear their credentials, especially on edit mode.
It might be annoying, but it isn't our main flow so I'm kind of ok with it for now.

I'll let others chime in to express their opinion here: @tinnytintin10 , @tehilashn , @JordanSh

Reply:
@mitodrummer how difficult it would be to clear the fields only on save instead? could we find OOTB hook we can use?

Reply:
As far as I'm aware, there is no "on save" hook we can detect to then apply our logic to clear other fields. Whatever the value of newPolicy is, will be what is sent when the user hits save. We could maybe wrap the root component of the fleet extension, and useState for a separate version of the policy with all creds present, and then send one out to fleet with the proper creds removed. It is a little messy but probably doable. It would be good to centralize this logic in one place, rather than replicate in both of those AWS credentials areas too. I'll reserve sometime today, to see if this is a viable approach.

Reply:
Just had my 1:1 with Kfir, it looks as if we could use the onPackagePolicyCreate and onPackagePolicyUpdate server side fleet hooks to mutate the policy just before it is saved. This might be a cleaner approach so we don't have to do anything funky with caching of credentials outside the policy on the frontend. I'll let ya know how this approach goes.